CRM/Activity/BAO/Activity.php

```diff
@@ -2140,6 +2140,10 @@ public static function getActivitySubject($entityObj) {
         $membershipType = CRM_Member_PseudoConstant::membershipType($entityObj->membership_type_id);
         $subject = $membershipType ? $membershipType : ts('Membership');
```

Comment:
We should use CRM_Core_Pseudoconstant::getKey & CRM_Member_BAO_Membership::buildOptions instead & deprecate that method. The ones I mention are highly standardised across multiple BAO (as is the api getoptions call)